modules/nw-http2-haproxy.adoc

```diff
@@ -0,0 +1,42 @@
+// Module included in the following assemblies:
+//
+// * networking/ingress-operator.adoc
+
+[id="nw-http2-haproxy{context}"]
+= HTTP/2 Ingress connectivity
+
+You can enable transparent end-to-end HTTP/2 connectivity in HAProxy. It allows application owners to make use of HTTP/2 protocol capabilities, including single connection, header compression, binary streams, and more.
+
+You can enable this HTTP/2 connectivity for an individual
+Ingress Controller or for the entire cluster.
+
+To enable the use of HTTP/2 for the connection from the client to HAProxy, a
+route must specify a custom certificate. A route that uses the default
+certificate cannot use HTTP/2. This restriction is necessary to avoid problems
+from connection coalescing, where the client re-uses a connection for different
+routes that use the same certificate.
+
+The connection from HAProxy to the application Pod can use HTTP/2 only for
+re-encrypt routes and not for edge-terminated or insecure routes. This
+restriction comes from the fact that HAProxy uses Application-Level Protocol
+Negotiation (ALPN), which is a TLS extension, to negotiate the use of HTTP/2
+with the backend. The implication is that end-to-end HTTP/2 is possible with
+passthrough and re-encrypt and not with insecure or edge-terminated routes.
+
+.Procedure
+
+Enable HTTP/2 on a single Ingress Controller.
+
+* To enable HTTP/2 on an Ingress Controller, run the `oc annotate` command:
++
+----
+$ oc -n openshift-ingress-operator annotate ingresscontrollers/<ingresscontroller_name> ingress.operator.openshift.io/default-enable-http2=true
+----
+
+Enable HTTP/2 on all Ingress Controllers.
```

Comment:
Clarify that this is for the entire cluster

Reply:
+!